src/Nethermind/Nethermind.Synchronization/FastBlocks/SyncStatusList.cs

```diff
@@ -43,7 +43,20 @@ public void GetInfosForBatch(BlockInfo?[] blockInfos)
                     switch (_statuses[currentNumber])
                     {
                         case FastBlockStatus.Unknown:
-                            blockInfos[collected] = _blockTree.FindCanonicalBlockInfo(currentNumber);
+                            BlockInfo? blockInfo = null;
+                            // Release the lock while performing the longer storage operation
+                            // to reduce lock contention
+                            Monitor.Exit(_statuses);
+                            try
+                            {
+                                blockInfo = _blockTree.FindCanonicalBlockInfo(currentNumber);
+                            }
+                            finally
+                            {
+                                // Re-enter the lock
+                                Monitor.Enter(_statuses);
+                            }
+                            blockInfos[collected] = blockInfo;
```

Comment:
This looks very hacky. Can we just accumulate those indexes into some collection and set `blockInfos[collected] = blockInfo;` with actually `FindCanonicalBlockInfo` outside the lock?